Compile on RedHat

When i compile on RedHat i have this exceptions:

Traceback (most recent call last):
  File "../deps/readies/bin/platform", line 20, in <module>
    platform = Platform()
  File "/home/sumame/RedisTimeSeries/deps/readies/paella/platform.py", line 25, in __init__
    Assert(False), "Cannot determine distribution"
NameError: name 'Assert' is not defined
Traceback (most recent call last):
  File "../deps/readies/bin/platform", line 20, in <module>
    platform = Platform()
  File "/home/sumame/RedisTimeSeries/deps/readies/paella/platform.py", line 25, in __init__
    Assert(False), "Cannot determine distribution"
NameError: name 'Assert' is not defined
Traceback (most recent call last):
  File "../deps/readies/bin/platform", line 20, in <module>
    platform = Platform()
  File "/home/sumame/RedisTimeSeries/deps/readies/paella/platform.py", line 25, in __init__
    Assert(False), "Cannot determine distribution"
NameError: name 'Assert' is not defined
Building rmutil...
Traceback (most recent call last):
  File "../../deps/readies/bin/platform", line 20, in <module>
    platform = Platform()
  File "/home/sumame/RedisTimeSeries/deps/readies/paella/platform.py", line 25, in __init__
    Assert(False), "Cannot determine distribution"
NameError: name 'Assert' is not defined
Traceback (most recent call last):
  File "../../deps/readies/bin/platform", line 20, in <module>
    platform = Platform()
  File "/home/sumame/RedisTimeSeries/deps/readies/paella/platform.py", line 25, in __init__
    Assert(False), "Cannot determine distribution"
NameError: name 'Assert' is not defined
Traceback (most recent call last):
  File "../../deps/readies/bin/platform", line 20, in <module>
    platform = Platform()
  File "/home/sumame/RedisTimeSeries/deps/readies/paella/platform.py", line 25, in __init__
    Assert(False), "Cannot determine distribution"
NameError: name 'Assert' is not defined
make[1]: Nothing to be done for `all'.
Linking ../bin/--release/redistimeseries.so...
/usr/lib/gcc/x86_64-redhat-linux/4.8.5/../../../../lib64/crt1.o: In function `_start':
(.text+0x20): undefined reference to `main'
../bin/--release/src/tsdb.o: In function `SeriesAddSample':
tsdb.c:(.text+0x469): undefined reference to `intrev64'
../bin/--release/src/tsdb.o: In function `SeriesQuery':
tsdb.c:(.text+0x506): undefined reference to `intrev64'
collect2: error: ld returned 1 exit status
make: *** [../bin/--release/redistimeseries.so] Error 1

--- a/deps/readies/bin/getredis5
+++ b/deps/readies/bin/getredis5
@@ -54,4 +54,4 @@ parser = argparse.ArgumentParser(description='Set up system for build.')
 parser.add_argument('-n', '--nop', action="store_true", help='no operation')
 args = parser.parse_args()
 
-RedisTimeSeriesSetup(nop = args.nop).setup()
+Redis5Setup(nop = args.nop).setup()

--- a/deps/readies/bin/platform
+++ b/deps/readies/bin/platform
@@ -6,7 +6,7 @@ import argparse
 
 READIES_PATH = os.path.realpath(os.path.join(os.path.dirname(__file__), ".."))
 sys.path.insert(0, READIES_PATH)
-from paella import Platform
+import paella
 
 parser = argparse.ArgumentParser(description='Report platform characteristics.')
 parser.add_argument('--os', action="store_true", help='Operating system')
@@ -15,9 +15,14 @@ parser.add_argument('--dist', action="store_true", help='Linux distribution (if 
 parser.add_argument('--arch', action="store_true", help='CPU Architecture')
 parser.add_argument('--kernel', action="store_true", help='Kernel version (if applicable)')
 parser.add_argument('--glibc', action="store_true", help='GLIBC version (if applicable)')
+parser.add_argument('--strict', action="store_true", help='Fail if cannot identify platform')
 args = parser.parse_args()
 
-platform = Platform()
+try:
+    platform = paella.Platform(strict=args.strict)
+except:
+    eprint("platform: cannot identify")
+    exit(1)
 ret = ""
 if args.os:
 	ret += " " + platform.os

--- a/deps/readies/paella/debug.py
+++ b/deps/readies/paella/debug.py
@@ -4,7 +4,10 @@ import os
 #----------------------------------------------------------------------------------------------
 
 if 'PYDEBUG' in os.environ:
-	from pdb import set_trace as bb
+    try:
+        from pudb import set_trace as bb
+    except ImportError:
+        from pdb import set_trace as bb
 else:
 	def bb(): pass
 

--- a/deps/readies/paella/files.py
+++ b/deps/readies/paella/files.py
@@ -1,13 +1,50 @@
 
 from contextlib import contextmanager
 import os
+import os.path
+import urllib2
+import tempfile
 
-def fread(fname, mode = 'rb'):
+#----------------------------------------------------------------------------------------------
+
+def fread(fname, mode='rb'):
 	with open(fname, mode) as file:
 		return file.read()
 
+#----------------------------------------------------------------------------------------------
+
+def fwrite(fname, text, mode='w'):
+	with open(fname, mode) as file:
+		return file.write(text)
+
+#----------------------------------------------------------------------------------------------
+
 def flines(fname, mode = 'rb'):
 	return [line.rstrip() for line in open(fname)]
+
+#----------------------------------------------------------------------------------------------
+
+def tempfilepath():
+    fd, path = tempfile.mkstemp()
+    os.close(fd)
+    return path
+
+#----------------------------------------------------------------------------------------------
+
+def wget(url, dest="", tempdir=False):
+    if dest == "":
+        dest = os.path.basename(url)
+        if dest == "":
+            dest = tempfilepath()
+        elif tempdir:
+            dest = os.path.join('/tmp', dest)
+    ufile = urllib2.urlopen(url)
+    data = ufile.read()
+    with open(dest, "wb") as file:
+        file.write(data)
+    return os.path.abspath(dest)
+
+#----------------------------------------------------------------------------------------------
 
 @contextmanager
 def cwd(path):
@@ -18,6 +55,10 @@ def cwd(path):
     finally:
         os.chdir(d0)
 
+#----------------------------------------------------------------------------------------------
+
 def mkdir_p(dir):
     if dir != '':
         os.makedirs(dir, exist_ok=True)
+
+#----------------------------------------------------------------------------------------------

--- a/deps/readies/paella/platform.py
+++ b/deps/readies/paella/platform.py
@@ -5,24 +5,59 @@ import platform
 #----------------------------------------------------------------------------------------------
 
 class Platform:
-    def __init__(self):
-        self.os = self.dist = self.os_ver = self.full_os_ver = self.os_nick = self.arch = '?'
-    
+    #------------------------------------------------------------------------------------------
+    class OSRelease():
+        def __init__(self):
+            self.defs = {}
+            with open("/etc/os-release") as f:
+                for line in f:
+                    try:
+                        k, v = line.rstrip().split("=")
+                        self.defs[k] = v.strip('"').strip("'")
+                    except:
+                        pass
+
+        def distname(self):
+            return self.defs["ID"].lower()
+
+        def version(self):
+            return self.defs["VERSION_ID"]
+
+        def osnick(self):
+            return self.defs["VERSION_CODENAME"]
+
+    #------------------------------------------------------------------------------------------
+
+    def __init__(self, strict=False):
+        self.os = self.dist = self.os_ver = self.full_os_ver = self.osnick = self.arch = '?'
+
         self.os = platform.system().lower()
-        dist = platform.linux_distribution()
-        distname = dist[0].lower()
-        self.os_ver = self.full_os_ver = dist[1]
         if self.os == 'linux':
+            if False:
+                dist = platform.linux_distribution()
+                distname = dist[0].lower()
+                self.os_ver = self.full_os_ver = dist[1]
+            else:
+                try:
+                    os_release = Platform.OSRelease()
+                    distname = os_release.distname()
+                    self.os_ver = self.full_os_ver = os_release.version()
+                except:
+                    if strict:
+                        assert(False), "Cannot determine distribution"
+                    distname = 'unknown'
+                    self.os_ver = self.full_os_ver = 'unknown'
             if distname == 'fedora' or distname == 'ubuntu' or  distname == 'debian' or distname == 'arch':
                 pass
-            elif distname == 'centos linux':
+            elif distname.startswith('centos'):
                 distname = 'centos'
-            elif distname.startswith('redhat'):
+            elif distname.startswith('redhat') or distname == 'rhel':
                 distname = 'redhat'
             elif distname.startswith('suse'):
                 distname = 'suse'
             else:
-                Assert(False), "Cannot determine distribution"
+                if strict:
+                    assert(False), "Cannot determine distribution"
             self.dist = distname
         elif self.os == 'darwin':
             self.os = 'macosx'
@@ -40,7 +75,10 @@ class Platform:
             self.os_ver = ''
             self.dist = ''
         else:
-            Assert(False), "Cannot determine OS"
+            if strict:
+                assert(False), "Cannot determine OS"
+            self.os_ver = ''
+            self.dist = ''
 
         self.arch = platform.machine().lower()
         if self.arch == 'amd64' or self.arch == 'x86_64':
@@ -54,7 +92,7 @@ class Platform:
 
     def is_debian_compat(self):
         return self.dist == 'debian' or self.dist == 'ubuntu'
-    
+
     def is_redhat_compat(self):
         return self.dist == 'redhat' or self.dist == 'centos'
 
@@ -86,12 +124,12 @@ class OnPlatform:
             self.common()
             if os == 'linux':
                 self.linux()
-                
+
                 if self.platform.is_debian_compat():
                     self.debian_compat()
                 if self.platform.is_redhat_compat():
                     self.redhat_compat()
-                
+
                 if dist == 'fedora':
                     self.fedora()
                 elif dist == 'ubuntu':
@@ -107,7 +145,7 @@ class OnPlatform:
                 elif dist == 'arch':
                     self.arch()
                 else:
-                    Assert(False), "Cannot determine installer"
+                    assert(False), "Cannot determine installer"
             elif os == 'macosx':
                 self.macosx()
 
@@ -130,13 +168,13 @@ class OnPlatform:
 
     def debian_compat(self): # debian, ubuntu, etc
         pass
-    
+
     def debian(self):
         pass
-    
+
     def centos(self):
         pass
-        
+
     def fedora(self):
         pass
 
@@ -145,7 +183,7 @@ class OnPlatform:
 
     def redhat(self):
         pass
-        
+
     def ubuntu(self):
         pass
 

--- a/deps/readies/paella/setup.py
+++ b/deps/readies/paella/setup.py
@@ -42,7 +42,7 @@ class RepoRefresh(OnPlatform):
 
     def redhat_compat(self):
         pass
-    
+
     def debian_compat(self):
         self.runner.run("apt-get -qq update -y")
 
@@ -60,7 +60,7 @@ class Setup(OnPlatform):
         self.os = self.platform.os
         self.dist = self.platform.dist
         self.ver = self.platform.os_ver
-        
+
         if self.has_command("python"):
             self.python = "python"
         elif self.has_command("python2"):
@@ -71,7 +71,7 @@ class Setup(OnPlatform):
         if self.os == 'macosx':
             # this is required because osx pip installed are done with --user
             os.environ["PATH"] = os.environ["PATH"] + ':' + '$HOME/Library/Python/2.7/bin'
-        
+
         if self.platform.is_debian_compat():
             # prevents apt-get from interactively prompting
             os.environ["DEBIAN_FRONTEND"] = 'noninteractive'
@@ -139,7 +139,7 @@ class Setup(OnPlatform):
         self.install(packs, group=True)
 
     #------------------------------------------------------------------------------------------
-    
+
     def yum_add_repo(self, repourl, repo=""):
         if not self.has_command("yum-config-manager"):
             self.install("yum-utils")
@@ -161,7 +161,7 @@ class Setup(OnPlatform):
 
     def pacman_add_repo(self, repourl, repo=""):
         pass
-    
+
     def brew_add_repo(self, repourl, repo=""):
         pass
 
@@ -209,3 +209,7 @@ class Setup(OnPlatform):
         if self.os == 'linux':
             self.install("ca-certificates")
         self.install("curl wget")
+
+    def install_git_lfs_on_linux(self):
+        self.run("curl -s https://packagecloud.io/install/repositories/github/git-lfs/script.deb.sh | bash")
+        self.install("git-lfs")

